src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -519,5 +486,152 @@ private void AddDirToListIfExist(List<string> dirs, string directory)
                 Log.LogMessage($"Fail to add directory: {directory} to the list because it doesn't exist.");
             }
         }
+#if net45
+        [DllImport("kernel32.dll", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        static extern bool DeleteFileW([MarshalAs(UnmanagedType.LPWStr)]string lpFileName);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool RemoveDirectory(string lpFileName);
+
+        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern IntPtr FindFirstFile(string lpFileName, out WIN32_FIND_DATA lpFindFileData);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        static extern bool FindNextFile(IntPtr hFindFile, out WIN32_FIND_DATA lpFindFileData);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        static extern bool FindClose(IntPtr hFindFile);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+        private static extern bool SetFileAttributesW(string lpFileName, FileAttributes dwFileAttributes);
+
+        // The CharSet must match the CharSet of the corresponding PInvoke signature
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        public struct WIN32_FIND_DATA
+        {
+            public uint dwFileAttributes;
+            public System.Runtime.InteropServices.ComTypes.FILETIME ftCreationTime;
+            public System.Runtime.InteropServices.ComTypes.FILETIME ftLastAccessTime;
+            public System.Runtime.InteropServices.ComTypes.FILETIME ftLastWriteTime;
+            public uint nFileSizeHigh;
+            public uint nFileSizeLow;
+            public uint dwReserved0;
+            public uint dwReserved1;
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
+            public string cFileName;
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
+            public string cAlternateFileName;
+        }
+
+        // Many Windows API's will use the unicode syntax when the path is prepended with '\\?\', this
+        // is instead of the ANSI paths which have a 260 character max path limit
+        private string MakeLongPath(string path)
+        {
+            if (path == null ||
+                path.Length == 0)
+            {
+                return path;
+            }
+            if (!path.StartsWith(@"\\?\"))
+            {
+                path = @"\\?\" + path;
+            }
+            return path;
+        }
+
+        private bool RemoveDirectoryRecursiveLongPath(string directory)
```

Comment:
You were able to create something longer then long path and successfully delete it with this, where it would normally fail to delete with the normal command line tools?